app/controllers/api/service_templates_controller.rb

```diff
@@ -4,5 +4,51 @@ class ServiceTemplatesController < BaseController
     include Subcollections::Tags
     include Subcollections::ResourceActions
     include Subcollections::ServiceRequests
+
+    def create_resource(_type, _id, data)
+      # Temporarily only supporting atomic.
+      raise 'Service Type composite not supported' if data['service_type'] == 'composite'
+      create_atomic(data).tap(&:save!)
+    rescue => err
+      raise BadRequestError, "Could not create Service Template - #{err}"
+    end
+
+    private
+
+    def create_atomic(data)
+      service_template = ServiceTemplate.new(data.except('request_info'))
+      if data.key?('request_info')
+        service_template.add_resource(create_service_template_request(data['request_info']))
+        set_new_resource_actions(data['request_info'], service_template)
+      end
+      service_template
+    end
+
+    # Need to set the request for non-generic Service Template
+    def create_service_template_request(request_data)
+      # data must be symbolized
+      request_params = request_data.symbolize_keys
+      wf = MiqProvisionWorkflow.class_for_source(request_params[:src_vm_id]).new(request_params, @auth_user_obj)
+      raise 'Could not find Provision Workflow class for source VM' unless wf
+      request = wf.make_request(nil, request_params)
+      raise 'Could not create valid request' if request == false || !request.valid?
+      request
+    end
+
+    # Set Resource Actions
+    def set_new_resource_actions(request_data, service_template)
+      dialog = request_data['dialog_id'].nil? ? nil : Dialog.find(request_data['dialog_id'])
+      [
+        {:name => 'Provision', :param_key => 'fqname'},
+        {:name => 'Reconfigure', :param_key => 'reconfigure_fqname'},
+        {:name => 'Retirement', :param_key => 'retire_fqname'}
+      ].each do |action|
+        unless request_data[action[:param_key]].nil?
```

Comment:
Even if the request_data does not have an action you'll still need to create an resource_action with default entry point. Look at `ServiceTemplate` and its subclasses for default entry points.